From dadf80ac8bf0cde1e5ac802dd78b1e7709e8ecf2 Mon Sep 17 00:00:00 2001
Message-Id: <dadf80ac8bf0cde1e5ac802dd78b1e7709e8ecf2.1421709247.git.chang-joon.lee@intel.com>
In-Reply-To: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
References: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
From: Shobhit Kumar <shobhit.kumar@intel.com>
Date: Wed, 17 Sep 2014 17:21:37 +0530
Subject: [PATCH 30/75] MUST_REBASE [VPG]: adf/intel: During interafce init
 Modelist was not populated

This cause usermode IOCTL calls to GET_DATA return empty modelist. call
pipe->ops->get_modelist to get the detailed list for userspace

Fixes issue in the basic intel-adf framework that came in from MCG and hence
better to have as separate patch for proper identification of framework
enhancements

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-3547
Change-Id: I5aa9118ae21db98efb0ed9b055bfe92eb9f418b2
Signed-off-by: Shobhit Kumar <shobhit.kumar@intel.com>
---
 drivers/video/adf/intel/intel_adf_interface.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/video/adf/intel/intel_adf_interface.c b/drivers/video/adf/intel/intel_adf_interface.c
index f50ba520..86873e1 100644
--- a/drivers/video/adf/intel/intel_adf_interface.c
+++ b/drivers/video/adf/intel/intel_adf_interface.c
@@ -427,6 +427,10 @@ int intel_adf_interface_init(struct intel_adf_interface *intf,
 	if (err)
 		goto out_err3;
 
+	/* fill the interface modelist */
+	pipe->ops->get_modelist(pipe, &intf->base.modelist,
+				&intf->base.n_modes);
+
 	/*turn on this interface if screen was connected*/
 	if (pipe->ops->is_screen_connected(pipe)) {
 		err = set_preferred_mode(intf);
-- 
1.7.9.5

